cmd/acra-keys/keys/migrate-keys.go

```diff
@@ -0,0 +1,285 @@
+func (m *MigrateKeysSubcommand) DryRun() bool {
+	return m.dryRun
+}
+
+// ForceWrite returns true if migration is allowed to overwrite existing destination key store.
+func (m *MigrateKeysSubcommand) ForceWrite() bool {
+	return m.force
+}
+
+// Name returns the same of this subcommand.
+func (m *MigrateKeysSubcommand) Name() string {
+	return CmdMigrateKeys
+}
+
+// GetFlagSet returns flag set of this subcommand.
+func (m *MigrateKeysSubcommand) GetFlagSet() *flag.FlagSet {
+	return m.flagSet
+}
+
+// RegisterFlags registers command-line flags of "acra-keys migrate".
+func (m *MigrateKeysSubcommand) RegisterFlags() {
+	m.flagSet = flag.NewFlagSet(CmdMigrateKeys, flag.ContinueOnError)
+	m.src.RegisterPrefixed(m.flagSet, "", "src_", "(old key store, source)")
+	m.dst.RegisterPrefixed(m.flagSet, "", "dst_", "(new key store, destination)")
+	m.flagSet.StringVar(&m.srcVersion, "src_keystore", "", "key store format to use: v1 (current), v2 (new)")
+	m.flagSet.StringVar(&m.dstVersion, "dst_keystore", "", "key store format to use: v1 (current), v2 (new)")
+	m.flagSet.BoolVar(&m.dryRun, "dry_run", false, "try migration without writing to the output key store")
+	m.flagSet.BoolVar(&m.force, "force", false, "write to output key store even if it exists")
+	m.flagSet.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Command \"%s\": migrate key store to a different format\n", CmdMigrateKeys)
+		fmt.Fprintf(os.Stderr, "\n\t%s %s [options...] \\\n"+
+			"\t\t--src_keystore <src-version> --src_keys_dir <.acrakeys-src> \\\n"+
+			"\t\t--dst_keystore <dst-version> --dst_keys_dir <.acrakeys-dst>\n",
+			os.Args[0], CmdMigrateKeys)
+		fmt.Fprintf(os.Stderr, "\nOptions:\n")
+		cmd.PrintFlags(m.flagSet)
+	}
+}
+
+// Parse command-line parameters of the subcommand.
+func (m *MigrateKeysSubcommand) Parse(arguments []string) error {
+	err := cmd.ParseFlagsWithConfig(m.flagSet, arguments, DefaultConfigPath, ServiceName)
+	if err != nil {
+		return err
+	}
+
+	if m.srcVersion == "" {
+		log.Warning("Missing required argument: --src_keystore={v1|v2}")
+	}
+	if m.dstVersion == "" {
+		log.Warning("Missing required argument: --dst_keystore={v1|v2}")
+	}
+	if m.srcVersion == "" || m.dstVersion == "" {
+		return ErrMissingFormat
+	}
+
+	if m.src.keyDir == "" {
+		log.Warning("Missing required argument: --src_keys_dir=<path>")
+	}
+	if m.dst.keyDir == "" {
+		log.Warning("Missing required argument: --dst_keys_dir=<path>")
+	}
+	if m.src.keyDir == "" || m.dst.keyDir == "" {
+		return ErrMissingKeyDir
+	}
+
+	if m.src.keyDirPublic == "" {
+		m.src.keyDirPublic = m.src.keyDir
+	}
+	if m.dst.keyDirPublic == "" {
+		m.dst.keyDirPublic = m.dst.keyDir
+	}
+
+	return nil
+}
+
+// Execute this subcommand.
+func (m *MigrateKeysSubcommand) Execute() {
+	if m.SrcKeyStoreVersion() == "v1" && m.DstKeyStoreVersion() == "v2" {
+		keyStoreV1, err := m.openKeyStoreV1(SrcMasterKeyVarName, m.SrcKeyStoreParams())
+		if err != nil {
+			log.WithError(err).Fatal("Failed to open keystore v1 (src)")
```

Comment:
let's use consistent wording: either `keystore` or `key store` everywhere?

(my personal pref is `keystore`, but i'm ok with both)

Reply:
It seems that both are used with similar frequency in strings, but code use is heavily skewed in favor of "keystore" as a single word because that's how the module is named.

Okay, I'll submit a separate PR to change it all into keystore. As much as I dislike such changes.